go/libraries/doltcore/sqle/index/index_lookup.go

```diff
@@ -115,15 +116,63 @@ func NewRangePartitionIter(ctx *sql.Context, t DoltTableable, lookup sql.IndexLo
 		nomsRanges:   nomsRanges,
 		prollyRanges: prollyRanges,
 		curr:         0,
-		durableState: durableState,
+		isDoltFmt:    isDoltFmt,
 	}, nil
 }
 
+func newPointPartitionIter(ctx *sql.Context, lookup sql.IndexLookup, idx *doltIndex) (sql.PartitionIter, error) {
+	tb := idx.keyBld
+	rng := lookup.Ranges[0]
+	ns := idx.ns
+	for j, expr := range rng {
+		v, err := getRangeCutValue(expr.LowerBound, rng[j].Typ)
+		if err != nil {
+			return nil, err
+		}
+		if err = PutField(ctx, ns, tb, j, v); err != nil {
+			return nil, err
+		}
+	}
+	tup := tb.BuildPermissive(sharePool)
+	return &pointPartitionIter{r: prolly.Range{Tup: tup, Desc: tb.Desc}}, nil
+}
+
+type pointPartitionIter struct {
```

Comment:
for every secondary lookup in a join we go from `sql.IndexLookup` -> `sql.PartitionIter` -> `sql.Partiiton` -> `sql.RowIter`. Is there any way to collapse that indirection, even if we keep the existing interfaces? Specifically I'm wondering if `pointPartitionIter` and `pointPartition` could be merged into one type

Reply:
Good call, that squeezed an extra 5% out of the join bench.